File: online-judge-solutions/Codeforces/1719B.cpp
//https://codeforces.com/problemset/problem/1719/B
//Lang		:	GNU C++14
//Time		:	343 ms
//Memory	:	800 KB
#include<iostream>
#include<vector>
#include<numeric>
using namespace std;
int main(){
	int t,n,k;
	cin>>t;
	while(t--){
		cin>>n>>k;
		vector<int> v(n);
		iota(begin(v),end(v),1);
		if(k%4==0) cout<<"NO\n";
		else{
			cout<<"YES\n";
			if(k%4==1 || k%4==3){
				for(int i=0;i<v.size();i+=2){
					cout<<v[i]<<" "<<v[i+1]<<endl;
				}
			}else{
				for(int i=0;i<v.size();i+=2){
					if(!((i/2)%2)) cout<<v[i+1]<<" "<<v[i]<<endl;
					else cout<<v[i]<<" "<<v[i+1]<<endl;
				}
			}
		}
	}
	return 0;
}

File: online-judge-solutions/Codeforces/1719B.py
#https://codeforces.com/problemset/problem/1719/B
#Lang		:	Python 3.8
#Time		:	390 ms
#Memory         :	7100 KB
for _ in range(int(input())):
    n,k=map(int,input().split())
    v=[i+1 for i in range(n)]
    if(k%4==0):print("NO")
    else:
        print("YES")
        if(k%4==1 or k%4==3):
            for i in range(0,len(v),2):
                print(v[i],v[i+1])
        else:
            for i in range(0,len(v),2):
                print(str(v[i+1])+" "+str(v[i]) if not (int(i/2))%2 else str(v[i])+" "+str(v[i+1]))

